src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs

```diff
@@ -46,6 +46,19 @@ internal static class FileRenameGenerator
             return allRenames;
         }
 
+        public static string ApplyRenameToPrimaryOutput(string primaryOutputPath, IEngineEnvironmentSettings environmentSettings, string sourceName, object resolvedNameParamValue, IParameterSet parameterSet, IReadOnlyDictionary<string, string> fileRenames, IReadOnlyList<IReplacementTokens> symbolBasedFileRenames = null)
+        {
+            IProcessor sourceRenameProcessor = SetupRenameProcessor(environmentSettings, fileRenames);
```

Comment:
can you confirm the error handling model here is to throw to the caller?  If not, would validating parameter state add value?